src/sync/sync_permission.hpp

```diff
@@ -93,12 +93,24 @@ class PermissionResults {
         return m_results.async(std::move(target));
     }
 
-    // Create a new instance by further filtering or sorting this instance.
+    // Create a new instance by further filtering this instance.
     PermissionResults filter(Query&& q) const
     {
         return PermissionResults(m_results.filter(std::move(q)));
     }
 
+    // Create a new instance by sorting this instance.
+    PermissionResults sort(SortDescriptor&& s) const
+    {
+        return PermissionResults(m_results.sort(std::move(s)));
+    }
+
+    // Get the results.
+    Results& results()
```

Comment:
You bring up good points. I'll go back to the Cocoa code and see if I can do it a different way first.